core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -2011,6 +2015,34 @@ public ImportDestinationArguments importDirectory(String directory) {
     return new BulkImport(directory, context);
   }
 
+  @Override
+  public TimeType getTimeType(final String tableName) throws TableNotFoundException {
+    if (tableName.equals(RootTable.NAME)) {
+      throw new IllegalArgumentException("accumulo.root table has no TimeType");
+    }
+    String systemTableToCheck =
+        MetadataTable.NAME.equals(tableName) ? RootTable.NAME : MetadataTable.NAME;
+    final Scanner scanner = context.createScanner(systemTableToCheck, Authorizations.EMPTY);
+    String tableId = tableIdMap().get(tableName);
+    if (tableId == null) {
+      throw new TableNotFoundException(null, tableName, "specified table does not exist");
+    }
+    final Text start = new Text(tableId);
+    final Text end = new Text(start);
+    start.append(new byte[] {'<'}, 0, 1);
+    end.append(new byte[] {'<'}, 0, 1);
+    scanner.setRange(new Range(start, end));
+    MetadataSchema.TabletsSection.ServerColumnFamily.TIME_COLUMN.fetch(scanner);
+    Entry<Key,Value> next = scanner.iterator().next();
+    Value val = next.getValue();
+    if (val.toString().startsWith("L")) {
+      return TimeType.LOGICAL;
+    } else if (val.toString().startsWith("M")) {
+      return TimeType.MILLIS;
+    }
+    throw new RuntimeException("Failed to retrieve TimeType");
```

Comment:
May be able to use ample for this.  Also may not need a special case for root tablet w/ ample, seems like it will return the time type logical.  But not sure, this is the [code that initializes the root tablet metadata](https://github.com/apache/accumulo/blob/114704bb58a37073a0401c3b6f479761089bd1fb/server/base/src/main/java/org/apache/accumulo/server/init/ZooKeeperInitializer.java#L189-L206) and makes me think it would return logical.

```suggestion
     String tableId = tableIdMap().get(tableName);
    if (tableId == null) {
      throw new TableNotFoundException(null, tableName, "specified table does not exist");
    }
        Optional<TabletMetadata> tabletMetadata = context.getAmple().readTablets().forTable(tableId)
            .fetch(ColumnType.TIME).checkConsistency().build().stream().findFirst();
    if(tabletMetadata.isPresent()) {
      return tabletMetadata.get().getTime().getType();
    } else {
      throw new RuntimeException("Failed to retrieve TimeType"))
    }
```

Reply:
@keith-turner thanks for the suggestion. I forgot about the use of ample. I made a few slight tweaks. 1) forTable requires a TableID vs a String, and 2) I updated to return a TimeType rather than a TableMetadata object.